packages/engine/Source/Core/PolygonGeometryLibrary.js

```diff
@@ -466,12 +469,294 @@ PolygonGeometryLibrary.polygonOutlinesFromHierarchy = function (
+    if (!followEdge) {
+      ++i;
+      continue;
+    }
+
+    if (!hasBeenVisited) {
+      // Wire another polygon starting at this position on the other side of the edge
+      polygonsToWire.push(i);
+    }
+
+    // Continue counter-clockwise to the next edge
+    const nextEdgeIndex = edgeIndex + (abovePlane ? 1 : -1);
+    const nextEdge = edgesOnPlane[nextEdgeIndex];
+    if (!defined(nextEdge)) {
+      ++i;
+      continue;
+    }
+
+    i = nextEdge.position;
+  } while (
+    i < positions.length &&
+    i >= 0 &&
+    i !== startIndex &&
+    polygon.length < positions.length
+  );
+
+  polygons.splice(polygonIndex, toDelete, polygon);
+
+  for (const index of polygonsToWire) {
+    polygonIndex = wirePolygon(
+      polygons,
+      ++polygonIndex,
+      positions,
+      edgesOnPlane,
+      0,
+      index,
+      !abovePlane
+    );
+  }
+
+  return polygonIndex;
+}
+
+/**
+ * Splits an array of polygons, defined as a list of Cartesian3 positions in counter-clockwise winding order, along the equator.
+ *
+ * @param {Array<Cartesian3[]>} outerRings An array of polygons, defined as a list of Cartesian3 positions in counter-clockwise winding order.
+ * @param {Ellipsoid} ellipsoid The ellipsoid to be used as a reference.
+ * @param {ArcType} arcType The type of line the polygon edges must follow. Valid options are {@link ArcType.GEODESIC} and {@link ArcType.RHUMB}.
+ * @param {Array<Cartesian3[]>} [result] An array of split polygons.
+ *
+ * @returns {Array<Cartesian3[]>} An array of split polygons.
+ */
+PolygonGeometryLibrary.splitPolygonsOnEquator = function (
+  outerRings,
+  ellipsoid,
+  arcType,
+  result
+) {
+  if (!defined(result)) {
+    result = [];
+  }
+
+  result.splice(0, 0, ...outerRings);
+  result.length = outerRings.length;
+
+  let currentPolygon = 0;
+  while (currentPolygon < result.length) {
+    // Adapted from https://www.sciencedirect.com/science/article/abs/pii/B9780125434577500589#:~:text=If%20the%20plane%20intersects%20the,tree%20and%20polygon%20intersection%20libraries
+    const outerRing = result[currentPolygon];
+    const positions = outerRing.slice();
+
+    if (outerRing.length < 3) {
+      result[currentPolygon] = positions;
+      ++currentPolygon;
+      continue;
+    }
+
+    // Step 1: Get all edges which intersect the split line, splciing any found intersections points into the list of positions
+    const edgesOnPlane = computeEdgesOnPlane(positions, ellipsoid, arcType);
+    // If nothing intersected (no point were added), or there is only a single point on the plane, use the original polygon
+    if (positions.length === outerRing.length || edgesOnPlane.length <= 1) {
+      result[currentPolygon] = positions;
+      ++currentPolygon;
+      continue;
+    }
+
+    // Step 2: Sort the edges along the split line by the distance between their starting points and the starting point of the split line.
+    edgesOnPlane.sort((a, b) => {
```

Comment:
Why do we need to sort here? (I don't have access to Vaněček's paper). In `wirePolygon`, we already do a `.findIndex()` to look up the appropriate edge.

Reply:
We sort the edges based on their signed distance along the splitting plane. In practice for us, this ends up being counter-clockwise along the equator. 

Having the edges in order ensures that we have the appropriate next edge spatially.

So once we find the index, `i`,  with `findIndex`, we now know that the spatially adjacent edge is at index `i+1`.